plotly/tools.py

```diff
@@ -1269,3 +1277,628 @@ def return_figure_from_figure_or_data(figure_or_data, validate_figure):
             )
 
     return figure
+
+
+class TraceFactory(dict):
+    @staticmethod
+    def create_quiver(x, y, u, v, scale=.1, arrow_scale=.3,
+                      angle=math.pi/9, **kwargs):
+        """
+        Return data for a quiver plot.
+
+        :param (list or array) x: x coordinates of the arrow locations
+        :param (list or array) y: y coordinates of the arrow locations
+        :param (list or array) u: x components of the arrow vectors
+        :param (list or array) v: y components of the arrow vectors
+        :param (float in [0,1]) scale: scales size of the arrows(ideally to
+            avoid overlap). Default = .1
+        :param (float in [0,1]) arrow_scale: value multiplied to length of barb
+            to get length of arrowhead. Default = .3
+        :param (angle in radians) angle: angle of arrowhead. Default = pi/9
+
+        :rtype (trace): returns quiver trace
+
+        :raises (PlotlyError): will happen if x, y, u, and v are not all
+            the same length (or size if ndarray)
+
+        :raises (PlotlyError): will happen if scale or arrow_scale are less
+            than or equal to 0
+
+        Example 1:
+        ```
+        # 1 Arrow from (0,0) to (1,1)
+        quiver = TraceFactory.create_quiver(x=[0], y=[0],
+                                            u=[1], v=[1],
+                                            scale=1)
+        # Plot
+        fig=Figure()
+        fig['data'].append(quiver)
+        py.iplot(fig, filename='quiver')
+        ```
+
+        Example 2:
+        ```
+        # Add data
+        x,y = np.meshgrid(np.arange(0, 2, .2), np.arange(0, 2, .2))
+        u = np.cos(x)*y
+        v = np.sin(x)*y
+
+        #Create quiver
+        quiver = TraceFactory.create_quiver(x, y, u, v)
+
+        # Plot
+        fig=Figure()
+        fig['data'].append(quiver)
+        py.iplot(fig, filename='quiver')
+        ```
+
+        Example 3:
+        ```
+        # Add data
+        x, y = np.meshgrid(np.arange(-np.pi, math.pi, .5),
+                           np.arange(-math.pi, math.pi, .5))
+        u = np.cos(x)*y
+        v = np.sin(x)*y
+
+        # Create quiver
+        quiver = TraceFactory.create_quiver(x, y, u, v, scale=.2,
+                                            arrow_scale=.3, angle=math.pi/6,
+                                            line=Line(color='purple', width=1))
+        # Plot
+        fig=Figure()
+        fig['data'].append(quiver)
+        py.iplot(fig, filename='quiver')
+        ```
+        """
+        barb_x, barb_y = Quiver(x, y, u, v, scale,
```

Comment:
@theengineear I totally agree: I started putting everything under TraceFactory and then worried it would be too massive/complicated with all of the chart types so I went back to organizing under separate classes.  `_Quiver` seems like a good solution.